apollo-router/src/query_planner/caching_query_planner.rs

```diff
@@ -326,20 +322,12 @@ where
                     }
                 };
 
-                let schema = self.schema.api_schema();
-                if let Ok(modified_query) = add_defer_labels(schema, &doc.ast) {
```

Comment:
this is already called inside BridgeQueryPlanner

Reply:
we could definitely rely on that. This is used when generating the HTTP response, to know which graphql error has to go in which deferred response (especially when there are multiple deferred fragments at the same level) https://github.com/apollographql/router/blob/ffec3f902bda97912929d8df9e0a447c936ac2ae/apollo-router/src/services/execution/service.rs#L376-L380

Reply:
I've filed a followup to do that as ROUTER-850